array/DNAm/preprocessing/CETYGOdeconvolution.r

```diff
@@ -0,0 +1,195 @@
+
+adultBrainCETYGO <- function(betas, cellType){
+  
+  predPropAll<-list()
+  CETYGOplots <- list()
+  propPlots <- list()
+  counter = 1
+  maxCETYGO <- 0
+  minCETYGO <- 1
+  
+  for(method in names(modelBrainCoef)){
+    for(j in 1:length(modelBrainCoef[[method]])){
+      if(!is.null(modelBrainCoef[[method]][[j]])){
+        predPropAll[[method]][[j]]<-projectCellTypeWithError(betas, modelBrainCoef[[method]][[j]])
+      }
+    } 
+  }
+  
+  
+  for(i in 1:length(predPropAll)){
+    type <- predPropAll[[1]]
+    for(j in 1:length(type)){
+      plotdf <- as.data.frame(type[[j]])
+      
+      if (ncol(plotdf) > 0) {
+        
+        if (max(plotdf$CETYGO) > maxCETYGO){
+          maxCETYGO <- max(plotdf$CETYGO)}
+        
+        if (min(plotdf$CETYGO) < minCETYGO){
+          minCETYGO <- min(plotdf$CETYGO)}
+      }
+    }
+  }
+  
+  for(i in 1:length(predPropAll)){
+    type <- predPropAll[[1]]
+    for(j in 1:length(type)){
+      plotdf <- as.data.frame(type[[j]])
+      
+      if (ncol(plotdf) > 0) {
+        
+        # CETYGO score boxplot
+        pCETYGO <- ggplot(plotdf, aes(factor(0), CETYGO))+
+          geom_boxplot()+
+          coord_cartesian(ylim=c(minCETYGO - 0.01, maxCETYGO + 0.01))+
+          ggtitle(paste0(cellType, "_", names(predPropAll[i]), "-", j))+
+          theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank(), plot.title = element_text(size=6))
+        
+        CETYGOplots[[counter]] <- pCETYGO
+        
+        # proportions boxplot
+        plotdf <-as.data.frame(plotdf[,1:(ncol(plotdf)-2)])
+        plotdf$Basename <- rownames(plotdf)
+        plotdf <- reshape2::melt(plotdf, id.vars = "Basename")
+        colnames(plotdf) <- c("Basename", "CellType", "Proportion")
+        
+        p <- ggplot(plotdf, aes(x=CellType, y = Proportion)) +
+          geom_boxplot()+
+          ggtitle(paste0(cellType, "_", names(predPropAll[i]), "-", j))+
+          theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
+        
+        propPlots[[counter]] <- p
+        
+        counter = counter + 1
+      }
+    }
+  }
+  
+  CETYGOplotFile <- paste0(dataDir, "/2_gds/QCmetrics/CETYGO/CETYGOscorePlots_", cellType, ".pdf")
+  pdf(CETYGOplotFile)
+  do.call(grid.arrange, c(CETYGOplots, ncol = 4))
+  dev.off()
+  
+  propPlotsFile <- paste0(dataDir, "/2_gds/QCmetrics/CETYGO/CETYGOcellPropsPlots_", cellType, ".pdf")
+  pdf(propPlotsFile, height = 20, width = 20)
+  do.call(grid.arrange, c(propPlots, ncol = 4))
+  dev.off()
+  
+  predPropOutFile <- paste0(dataDir, "/2_gds/QCmetrics/CETYGO/CETYGOpredictedProportions_", cellType, ".rdat")
+  save(predPropAll, file = predPropOutFile)
+  
+}
+
+
+#----------------------------------------------------------------------#
+# RUN CETYGO FUNCTION ON EACH CELL TYPE
+#----------------------------------------------------------------------#
+
+for(cell in sampleSheet$Cell_Type){
```

Comment:
Only relevant for datasets with diff cell types. Again need to check config file variable before execution

Reply:
Do we also want a figure of this? 
Where/how will we incorporate this as a QC check?

Reply:
Have edited to include option for bulk/blood data

QC check - threshold parameter in config.r? 0.07? Move to calcQCmetrics and add CETYGO score as variable there?